src/Core/Settings.h

```diff
@@ -306,6 +306,7 @@ struct Settings : public SettingsCollection<Settings>
     \
     M(SettingUInt64, max_memory_usage, 0, "Maximum memory usage for processing of single query. Zero means unlimited.", 0) \
     M(SettingUInt64, max_memory_usage_for_user, 0, "Maximum memory usage for processing all concurrently running queries for the user. Zero means unlimited.", 0) \
+    M(SettingUInt64, max_memory_usage_for_session, 0, "Maximum memory usage for processing all concurrently running queries for a user session. Zero means unlimited.", 0) \
```

Comment:
It's unclear what is the motivation of this setting.
Is it for Memory and Temporary tables?

Reply:
Yes, this is required when there are multiple temp tables involved, so sessions can easily get out of user and query limits.